### ISSUE TITLE
Handling time in string format

### ISSUE THREAD

**Issue by SarahAlidoost** [eWaterCycle/ewatercycle#90]
In ewatercycle, the start/end time is given as a string in the [ISO format](https://nl.wikipedia.org/wiki/ISO_8601) (which uses the Grogerian calendar), and in UTC time. A date string ISO format is represented differently by different libraries:

- **[datetime.datetime](https://docs.python.org/3/library/datetime.html)**
```python
from datetime import datetime, timezone
datetime.fromisoformat('2000-01-01T00:00:00+00:00')
#out: datetime.datetime(2000, 1, 1, 0, 0, tzinfo=datetime.timezone.utc)
datetime(2000,1, 1, tzinfo=timezone.utc).isoformat()
#out: '2000-01-01T00:00:00+00:00'
```

- **[numpy.datetime64](https://numpy.org/doc/stable/reference/arrays.datetime.html)**
```python
import numpy as np
np.datetime64('2000-01-01T00:00:00.0')
#out: numpy.datetime64('2000-01-01T00:00:00.000')
np.datetime_as_string(np.array(np.datetime64('2000-01-01T00:00:00.0')), timezone='UTC')
#out: '2000-01-01T00:00:00.000Z'
```

- **[pandas.date_range](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.date_range.html)**
```python
import pandas as pd
pd.date_range(start='1/1/2000', periods=1, tz='UTC')
#out: DatetimeIndex(['2000-01-01 00:00:00+00:00'], dtype='datetime64[ns, UTC]', freq='D')
```

- **[cftime.datetime](https://unidata.github.io/cftime/api.html)**
```python
import cftime
cftime.datetime(2000,1, 1)
#out: cftime.datetime(2000, 1, 1, 0, 0, 0, 0)
cftime.datetime(2000,1, 1).isoformat()
#out: '2000-01-01T00:00:00'
```

- **xarray with np.datetime64**
```python
import numpy as np
import xarray as xr
da = xr.DataArray(
        [0.1],
        coords=[[np.datetime64('2000-01-01')]],
        dims=['time'],
        )
da.coords['time'][0]
#out: array('2000-01-01T00:00:00.000000000', dtype='datetime64[ns]')
np.datetime_as_string(da.coords['time'][0], timezone='UTC')
#out: '2000-01-01T00:00:00.000000000Z'
```

- **xarray with datetime**
```python
from datetime import datetime, timezone
import xarray as xr
da = xr.DataArray(
        [0.1],
        coords=[[datetime(2000, 1, 1, tzinfo=timezone.utc)]],
        dims=['time'],
        )
da.coords['time'][0]
#out: array('2000-01-01T00:00:00.000000000', dtype='datetime64[ns]')
np.datetime_as_string(da.coords['time'][0], timezone='UTC')
#out: '2000-01-01T00:00:00.000000000Z'
```

**Comment by SarahAlidoost** [comment]
I suggest using the numpy iso format i.e. '2000-01-01T00:00:00Z' and library `dateutil` to parse it as:

```python
from dateutil.parser import parse
parse('2000-01-01T00:00:00Z')
#out: datetime.datetime(2000, 1, 1, 0, 0, tzinfo=tzutc())
```

and to convert the datetime object back to the iso format:
```python
parse('2000-01-01T00:00:00Z').strftime("%Y-%m-%dT%H:%M:%SZ")
#out: '2000-01-01T00:00:00Z'
```

**Comment by Peter9192** [comment]
Nice overview! So the internal representation of dateutil is 'normal' `datetime.datetime` objects?

**Comment by SarahAlidoost** [comment]
> Nice overview! So the internal representation of dateutil is 'normal' `datetime.datetime` objects?

It seems so. It is mentioned in the [documentation](https://dateutil.readthedocs.io/en/stable/) that "the dateutil module provides powerful extensions to the standard datetime module".

**Comment by Peter9192** [comment]
Addressed in #137
